src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -615,235 +570,47 @@ sendMoney
     -> MoneySource
     -> NonEmpty (CId Addr, Coin)
     -> m CTx
-sendMoney SendActions {..} passphrase moneySource dstDistr = do
-    (spendings, outputs) <- prepareTx passphrase moneySource dstDistr
-    sendDo spendings outputs
+sendMoney SendActions{..} passphrase moneySource dstDistr = do
+    addrMetas' <- getMoneySourceAddresses moneySource
+    addrMetas <- nonEmpty addrMetas' `whenNothing`
+        throwM (RequestError "Given money source has no addresses!")
+    sks <- forM addrMetas $ getSKByAccAddr passphrase
+    srcAddrs <- forM addrMetas $ decodeCIdOrFail . cwamId
+
+    withSafeSigners passphrase sks $ \ss -> do
+        let hdwSigner = NE.zip ss srcAddrs
+        relatedAccount <- getSomeMoneySourceAccount moneySource
+        outputs <- coinDistrToOutputs dstDistr
+        TxAux {taTx = tx} <- rewrapTxError "Cannot send transaction" $
+            submitMTx enqueueMsg hdwSigner outputs (relatedAccount, passphrase)
+
+        ownUtxo <- getOwnUtxos $ toList srcAddrs
```

Comment:
Why do you do it just after you created transaction which seem to remove inputs you seek?